### DISCUSSION TITLE
Multi Tenancy with Raven

### DISCUSSION THREAD

**Discussion by BernhardGlueck** [ravendb/ravendb#13162] [Q&A] [answered]
Hi !

I am currently working on a SAAS solution with RavenDB as its primary database. 
For simplicity reasons i would prefer the "one database per tenant" approach. 
The whole application is mainly an ASP.NET Core Web API.... 

My question is if i should create one IDocumentStore per Tenant, or reuse the same one and wrap session handling with some tenancy logic.... I am not sure if DocumentStores are really independent or if they cache some data globally ... 

( I have per tenant service scopes already so creating an IDocumentStore per tenant would be trivial )

**Comment by ayende** [comment, marked as answer]
The cache we have is done at the `RequestExecuter` level, which is a *per database* value. Note that the `MaxHttpCacheSize` in the convention refers to _each_ cache, not global across all of them.

In other words, each database, either using a single document store or using separate document stores will have its own cache.

**Reply by BernhardGlueck** [reply]
Got it, in that case I will just create a DocumentStore per Tenant
